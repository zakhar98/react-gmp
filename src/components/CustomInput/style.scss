.input {
  &__label {
    color: #f65261;
    text-transform: uppercase;
    font-size: 18px;
    font-weight: 500;
  }

  &__field {
    &.error {
      border: 2px solid #f65261;
    }

    width: 100%;
    font-family: 'Work Sans', sans-serif;
    font-weight: 400;
    padding: 12px;
    margin-top: 8px;
    border: none;
    border-radius: 4px;
    color: white;
    font-size: 18px;
    background-color: rgba(255, 255, 255, .1);
    outline: none;
  }

  &__field[type="date"]::-webkit-calendar-picker-indicator {
    cursor: pointer;
    border-radius: 4px;
    margin-right: 2px;
    opacity: 0.6;
    filter: invert(0.8);
  }
  
  &__field[type="date"]::-webkit-calendar-picker-indicator:hover {
    opacity: 1
  }

  &__error-text {
    color: #f65261;
    font-size: 12px;
    padding-top: 4px;
  }
}
