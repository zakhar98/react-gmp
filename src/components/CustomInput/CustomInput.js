import React from "react";
import PropTypes from "prop-types";
import "./style.scss";

export default function CustomInput(
  {name, type, value, label, id, placeholder, onChange}) {
    return (
      <div className="input">
        <label className="input__label" htmlFor={id}>{label}</label><br/>
        <input className="input__field"
          type={type}
          id={id}
          name={name}
          value={value}
          placeholder={placeholder}
          onChange={onChange}
        />
      </div>
    );
}

CustomInput.propTypes = {
  name: PropTypes.string,
  type: PropTypes.string,
  value:  PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number
  ]),
  label: PropTypes.string,
  id: PropTypes.string,
  placeholder: PropTypes.string,
  onChange: PropTypes.func,
};
