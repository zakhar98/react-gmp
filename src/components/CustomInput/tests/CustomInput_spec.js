import React from 'react';
import renderer from 'react-test-renderer';
import CustomInput from '../CustomInput.js';
import * as formik from 'formik';

formik.useField = jest.fn(()=>[{}, {}, {}]);

describe('CustomInput', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<CustomInput />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
