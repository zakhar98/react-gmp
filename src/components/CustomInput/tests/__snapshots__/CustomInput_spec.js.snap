// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput renders correctly 1`] = `
<div
  className="input"
>
  <label
    className="input__label"
  />
  <br />
  <input
    className="input__field "
    onBlur={[Function]}
  />
</div>
`;
